Initial work supporting generic EventGridEvent<T>

drub0y opened the pull request:
Fixes #11.

drub0y committed:
Initial work supporting generic EventGridEvent<T>

dnfclas commented:
@__drub0y__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla2.dotnetfoundation.org.

It will cover your contributions to all .NET Foundation-managed open source projects.
_Thanks,
.NET Foundation Pull Request Bot_

drub0y committed:
Add param validation to EventGridTriggerBinding

This adds parameter validation to the
`EventGridTriggerBinding::BindAsync` implementation and also refactors
the unit tests that were added to the previous commit to align with the
other changes to add support for the generic `EventGridEvent<T>`.

drub0y committed:
Cleaning up unit tests

Improving unit tests to better utilitze XUnit assertion library when
waiting for async exceptions with `ThrowsAsync`.
